### PR TITLE
PL-1189 - adding get balance list api call

### PR DESCRIPTION
**Pull request by VKCR**

### CODE REVIEW

File: lib/src/main/java/io/token/Member.java [hunk 1]
```diff
@@ -843,6 +843,32 @@ public Money getCurrentBalance(String accountId, Key.Level keyLevel) {
...
     }
 
+    /**
+     * Looks up current balance for a list of accounts.
+     *
+     * @param accountIds list of account ids
+     * @param keyLevel key level
+     * @return map of account id to current balance
+     */
+    public Map<String, Money> getCurrentBalanceList(
```
Comment (ab666) [inline lib/src/main/java/io/token/Member.java]: why do you call it getCurrentBalance**List** if you are returning a **Map**?

File: lib/src/main/java/io/token/rpc/Client.java [hunk 2]
```diff
@@ -822,6 +828,150 @@ public Money apply(GetBalanceResponse response) {
...
     }
 
+    /**
+     * Get the current balance of a list of accounts.
+     *
+     * @param accountIdList list of account ids
+     * @param keyLevel key level
+     * @return map of account ids to GetBalanceResponses
+     */
+    public Observable<Map<String, Money>> getCurrentBalanceList(
```
Comment (ab666) [inline lib/src/main/java/io/token/rpc/Client.java]: this is a pretty ugly method :)

File: lib/src/main/java/io/token/rpc/Client.java [hunk 3]
```diff
@@ -822,6 +828,150 @@ public Money apply(GetBalanceResponse response) {
...
+                    public Map<String, Money> apply(GetBalanceListResponse response) {
+                        return response
+                                .getResponseMap()
+                                .entrySet()
+                                .stream()
+                                .filter(new Predicate<Map.Entry<String, GetBalanceResponse>>() {
+                                    @Override
+                                    public boolean test(
+                                            Map.Entry<String, GetBalanceResponse> entry) {
+                                        return entry.getValue().getStatus() == SUCCESSFUL_REQUEST
```
Comment (ab666) [inline lib/src/main/java/io/token/rpc/Client.java]: same here

File: lib/src/main/java/io/token/Member.java (hunk 1 above)
Comment (VKCR) [reply lib/src/main/java/io/token/Member.java]: You're right, I'll change everything to ...BalanceMap

File: lib/src/main/java/io/token/rpc/Client.java (hunk 2 above)
Comment (VKCR) [reply lib/src/main/java/io/token/rpc/Client.java]: No lambda expressions...

**Comment by ab666** [comment]
would require changes based on the service side review.

**Comment by ab666** [comment]
Your commit message ("making changes") is not very descriptive

File: lib/src/main/java/io/token/rpc/Client.java [hunk 4]
```diff
@@ -749,35 +749,25 @@ public Boolean apply(GetDefaultAccountResponse response) {
...
                 .getBalance(GetBalanceRequest
                         .newBuilder()
-                        .setPayload(payload)
-                        .setSignature(Signature
-                                .newBuilder()
-                                .setMemberId(memberId)
-                                .setKeyId(signer.getKeyId())
-                                .setSignature(signer.sign(payload)))
+                        .setAccountId(accountId)
+                        .setNonce(UUID.randomUUID().toString())
```
Comment (ab666) [inline lib/src/main/java/io/token/rpc/Client.java]: Why is it needed? can't we just use the same nonce of the request signature?

Comment (VKCR) [reply lib/src/main/java/io/token/rpc/Client.java]: You mean use the request signature as the nonce?

**Pull request merged**